xc/device.ml

```diff
@@ -1819,19 +1820,29 @@ module Dm_Common = struct
     ; fd_map = []
     }
 
-  let vgpu_args_of_nvidia domid vcpus (vgpu:Xenops_interface.Vgpu.nvidia) pci restore device =
+  let vgpu_args_of_nvidia domid vcpus vgpus restore =
     let open Xenops_interface.Vgpu in
+    let virtual_pci_address_compare vgpu1 vgpu2 =
+      match vgpu1, vgpu2 with
+      |{implementation = Nvidia {virtual_pci_address = Some pci1; _}}, {implementation = Nvidia {virtual_pci_address = Some pci2; _}} -> Pervasives.compare pci1.dev pci2.dev
```

Comment:
Please avoid overlong lines. These are painful to review on GitHub.

Reply:
I would prefer something like the code below which provides still a total order but orders them based on the component that you care about. I think an order relation not raise an exception.
```
let compare t1 t2 = match t1, t2 with
  | {implementation = Nvidia {virtual_pci_address = Some pci1; _}; _},
    {implementation = Nvidia {virtual_pci_address = Some pci3; _}; _} ->
    Pervasives.compare pci1, pci2
  | other1, other2 -> Pervasives.compare other1 other2
```

Reply:
It looks graceful.